test/async_utils_test.py

```diff
@@ -624,3 +628,70 @@ async def foo():
     async for item in callable_to_agen(foo):
         result.append(item)
     assert result == [await foo()]
+
+
+def test_sigint_run_async_gen_shuts_down_gracefully():
+    code = textwrap.dedent(
+        """
+    import asyncio
+    import time
+    from itertools import count
+    from modal._utils.async_utils import Runner, aclosing
+    async def async_gen():
+        print("enter")
+        try:
+            for i in count():
+                yield i
+                await asyncio.sleep(0.1)
+        finally:
+            # this could be either CancelledError or GeneratorExit depending on timing
+            # CancelledError happens if sigint is during this generator's await
+            # GeneratorExit is during the yielded block in the sync caller
+            print("cancel")
+            await asyncio.sleep(0.1)
+            print("bye")
+    try:
+        with Runner() as runner:
+            for res in runner.run_async_gen(async_gen()):
+                print("res", res)
+    except KeyboardInterrupt:
+        print("KeyboardInterrupt")
+    """
+    )
+    if sys.platform == "win32":
```

Comment:
This is probably what I spent the most time on - getting the Ctrl-C test to work on windows... Had to fire up the ol' gaming PC to run som tests...

Reply:
![image](https://github.com/user-attachments/assets/cbdd89a4-8a4c-4d66-8422-13e255c4a083)